src/artifact_choosing.rs

```diff
@@ -1,8 +1,11 @@
 #[cfg(target_os = "windows")]
 static PLATFORM_KEYWORDS: &[&str] = &["win32", "win64", "windows"];
 
-#[cfg(target_os = "macos")]
-static PLATFORM_KEYWORDS: &[&str] = &["macos", "darwin"];
+#[cfg(all(target_os = "macos", target_arch = "x86_64"))]
+static PLATFORM_KEYWORDS: &[&str] = &["macos-x86_64", "darwin-x86_64", "macos", "darwin"];
+
+#[cfg(all(target_os = "macos", target_arch = "aarch64"))]
+static PLATFORM_KEYWORDS: &[&str] = &["macos-arm64", "darwin-arm64", "macos-x86_64", "darwin-x86_64", "macos", "darwin"];
```

Comment:
I'm assuming some of these won't work for M1?

Reply:
x86_64 apps work with Rosetta2 (with caveats), which most users will have installed, because we are in the transitory period right now https://support.apple.com/en-us/HT211861 
The primary caveat is that dependencies installation can be tricky for x86_64. E.g. if x86_64 binary depends on OpenSSL, user would need to install x86_64 version of OpenSSL.
Thus we should prefer to use arm binaries.
If x86_64 binary has no runtime dependecies, it should work just fine if Rosetta2 is installed

Reply:
In that case I think it's all good because foreman will try to find the first asset in that list, which will make it prioritize the correct specific binaries. Thanks for your contribution 👍